toolium/utils/dataset.py

```diff
@@ -308,30 +308,29 @@ def _replace_param_fixed_length(param):
     :param param: parameter value
     :return: tuple with replaced value and boolean to know if replacement has been done
     """
+    STRING_SEED = 'a'
+    INTEGER_SEED = '1'
     new_param = param
     param_replaced = False
+    # we allow partial replacements for STRING
+    string_expression = re.compile(r'\[STRING_WITH_LENGTH_([0-9]+)\]')
+    new_param = string_expression.sub(lambda x: int(x.group(1)) * 'a', param)
+
+
     if param.startswith('[') and param.endswith(']'):
         if any(x in param for x in ['STRING_ARRAY_WITH_LENGTH_', 'INTEGER_ARRAY_WITH_LENGTH_']):
-            seeds = {'STRING': 'a', 'INTEGER': 1}
+            seeds = {'STRING': STRING_SEED, 'INTEGER': int(INTEGER_SEED)}
             seed, length = param[1:-1].split('_ARRAY_WITH_LENGTH_')
             new_param = list(seeds[seed] for x in range(int(length)))
             param_replaced = True
         elif 'JSON_WITH_LENGTH_' in param:
             length = int(param[1:-1].split('JSON_WITH_LENGTH_')[1])
             new_param = dict((str(x), str(x)) for x in range(length))
             param_replaced = True
-        elif any(x in param for x in ['STRING_WITH_LENGTH_', 'INTEGER_WITH_LENGTH_']):
-            seeds = {'STRING': 'a', 'INTEGER': '1'}
-            # The chain to be generated can be just a part of param
```

Comment:
this could not be because we are inside an if that checks that param starts and ends with []